### PR TITLE
added Watchlist and Favorites in the TMDB definitions

### PR DESCRIPTION
**Pull request by elChapoSing**
had to add account_id field to the UI for the v4 tmdb api

#### Database Migration
NO

#### Todos
- [x] Tests
- [ ] Fill in the Radarr API Read Access Token (v4 auth)
- [ ] Integrate getting the account ID from within Radarr ? (see https://github.com/elChapoSing/tmdb_account_id)

#### Issues Fixed or Closed by this PR
https://github.com/Radarr/Radarr/issues/3496
* #

### CODE REVIEW

**Comment by elChapoSing** [comment]
No interest on this ?

**Comment by tkdrob** [comment]
The more features, the better. It's just not my call. :)

Waiting on my own pull request, too

**Comment by elChapoSing** [comment]
Am a bit disappointed :) thought that was a highly sought after feature...

File: src/NzbDrone.Core/NetImport/TMDb/TMDbRequestGenerator.cs [hunk 1]
```diff
@@ -72,6 +79,7 @@ public virtual NetImportPageableRequestChain GetMovies()
                 requestBuilder.Resource(tmdbParams);
 
                 var request = requestBuilder
+                    .SetHeader("Bearer","NEED RADARR API Read Access Token (v4 auth) HERE")
```
Comment (galli-leo) [inline src/NzbDrone.Core/NetImport/TMDb/TMDbRequestGenerator.cs]: Maybe I misunderstood the documentation, but isn't this specific to each user? So wouldn't this need to come from the user?

File: src/NzbDrone.Core/NetImport/TMDb/TMDbSettings.cs [hunk 2]
```diff
@@ -69,6 +69,10 @@ public TMDbSettings()
         [FieldDefinition(0, Label = "TMDb API URL", HelpText = "Link to to TMDb API URL, do not change unless you know what you are doing.")]
         public string Link { get; set; }
 
+        [FieldDefinition(0, Label = "Account ID", HelpText = "Your account ID for Radarr API.")]
+//        I have built a simple tool at https://github.com/elChapoSing/tmdb_account_id to get it but not sure how to integrate it.
```
Comment (galli-leo) [inline src/NzbDrone.Core/NetImport/TMDb/TMDbSettings.cs]: You could include a link to something with more information (e.g. a wiki page).

File: src/NzbDrone.Core/NetImport/TMDb/TMDbSettings.cs [hunk 3]
```diff
@@ -69,6 +69,10 @@ public TMDbSettings()
         [FieldDefinition(0, Label = "TMDb API URL", HelpText = "Link to to TMDb API URL, do not change unless you know what you are doing.")]
         public string Link { get; set; }
 
+        [FieldDefinition(0, Label = "Account ID", HelpText = "Your account ID for Radarr API.")]
```
Comment (galli-leo) [inline src/NzbDrone.Core/NetImport/TMDb/TMDbSettings.cs]: Field Definitions cannot have the same number! I would move this down to be the last item. At least further down than the list type.

File: src/NzbDrone.Core/NetImport/TMDb/TMDbRequestGenerator.cs (hunk 1 above)
Comment (galli-leo) [reply src/NzbDrone.Core/NetImport/TMDb/TMDbRequestGenerator.cs]: Also, if this is not the case, wouldn't we still need an access_token for each user to get those lists? Something we can only do by redirecting them to the TMDB homepage?

File: src/NzbDrone.Core/NetImport/TMDb/TMDbSettings.cs (hunk 3 above)
Comment (elChapoSing) [reply src/NzbDrone.Core/NetImport/TMDb/TMDbSettings.cs]: thought I fixed that one - will change it.

File: src/NzbDrone.Core/NetImport/TMDb/TMDbRequestGenerator.cs (hunk 1 above)
Comment (elChapoSing) [reply src/NzbDrone.Core/NetImport/TMDb/TMDbRequestGenerator.cs]: No that one is the generic Radarr one, only the V4 version. Each user could use its own for sure but the way it's set up currently for v3 in radarr is everyone uses the radarr one. Guess it makes it easier for you to track usage stats etc.

File: src/NzbDrone.Core/NetImport/TMDb/TMDbSettings.cs (hunk 2 above)
Comment (elChapoSing) [reply src/NzbDrone.Core/NetImport/TMDb/TMDbSettings.cs]: ideally Radarr should include a way to get the account_id.
Yes you need to have the user identify on tmdb.
My repo demonstrates the proper way to get it based on tmdb v4 documentation (https://developers.themoviedb.org/4/auth/user-authorization-1) but properly integrating it into Radarr is beyond my skill level I'm afraid.
If you think adding it as a parameter that the user has to fill is enough then just linking to my repo in the help tooltip should be enough ? I will make sure the readme there is well laid out.

**Comment by elChapoSing** [comment]
will wait for your comments on my replies before doing anything more

File: src/NzbDrone.Core/NetImport/TMDb/TMDbRequestGenerator.cs (hunk 1 above)
Comment (galli-leo) [reply src/NzbDrone.Core/NetImport/TMDb/TMDbRequestGenerator.cs]: API token is `eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIxYTczNzMzMDE5NjFkMDNmOTdmODUzYTg3NmRkMTIxMiIsInN1YiI6IjU4NjRmNTkyYzNhMzY4MGFiNjAxNzUzNCIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.gh1BwogCCKOda6xj9FRMgAAj_RYKMMPC3oNlcBtlmwk`

File: src/NzbDrone.Core/NetImport/TMDb/TMDbSettings.cs [hunk 4]
```diff
@@ -64,6 +64,7 @@ public TMDbSettings()
             MinVoteAverage = "5";
             MinVotes = "1";
             LanguageCode = (int)TMDbLanguageCodes.en;
+            AccountID = "https://github.com/elChapoSing/tmdb_account_id";
```
Comment (galli-leo) [inline src/NzbDrone.Core/NetImport/TMDb/TMDbSettings.cs]: Can you move this link to the FieldDefinition? There is a HelpLink property on that, which will automatically create a hyperlink.

**Comment by elChapoSing** [comment]
think should be ok now.

**Comment by elChapoSing** [comment]
Anything else @galli-leo @Radarr ?

**Comment by elChapoSing** [comment]
ready to actually pull it in next release ? @galli-leo @Radarr

**Comment by skuid87** [comment]
Hey guys - what's the latest with this? Is there a working method to add TMDb watchlists/favourites lists into Radarr now?

**Comment by Qstick** [comment]
We are eyeing this for Aphrodite, develop is mainly getting bug fixes only

**Comment by skuid87** [comment]
> We are eyeing this for Aphrodite, develop is mainly getting bug fixes only

Thanks for the reply. This might be a stupid question, but is there an approximate timeline for Aphrodite release? Just wondering if I should wait, or seek an alternative method to get my current Plex library into Radarr.

**Comment by Qstick** [comment]
It’s in open alpha now, no specific timeframe for a more stable release of it.

**Pull request closed without merging**